### PR TITLE
New verbs and fold refactor (UI and core)

### PR DESCRIPTION
**Pull request by gaudyb**
- Core implementation and unit testing for new verbs: erase, merge, pivot and unfold
- React component implementation for new verbs: erase, merge, pivot and unfold
- Refactor of fold verb to receive custom names for key and value columns
- Spread fix to use custom output columns

### CODE REVIEW

**Comment by natoverse** [comment]
* Spread component: It would help to have a label above the optional output columns to differentiate from the spread column (i.e., "New column names").
* Spread component: Editing the output columns doesn't quite work - as you type the textbox loses focus after each letter. The output is correct though.
* Spread typing: can we change the output `columns` name to `to` so it is consistent with our other verbs for output column naming (`columns` is used elsewhere for inputs, so this could be confusing).
* Can you add an example of all new verbs to the `every-operation.json` config example in the webapp, so we have examples of all new verbs

**Review by natoverse — CHANGES_REQUESTED** [review]

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ import { TableStore } from './TableStore.js'
  */
 export type Value = any
 
+export enum MergeStrategy {
```
[the remark this answers is not in the document]
Comment (gaudyb) [reply packages/core/src/types.ts:13]: Done

File: packages/core/src/engine/verbs/pivot.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @param step
+ * @param store
+ * @returns
+ */
+export async function pivot(
+	step: Step,
+	store: TableStore,
+): Promise<ColumnTable> {
+	const { input, args } = step
+	const { key, value, operation } = args as PivotArgs
```
[the remark this answers is not in the document]
Comment (gaudyb) [reply packages/core/src/engine/verbs/pivot.ts:23]: Done

**Review by natoverse — APPROVED** [review]

File: packages/react/src/compounds/FilterAggregateLookup/FilterAggregateLookup.tsx [hunk 3]
```diff
@@ -68,7 +68,8 @@ export const FilterAggregateLookup: React.FC<StepComponentProps> = memo(
 		)
 
 		const handleToChange = useCallback(
-			(e, to) => {
+			(_e, to) => {
+				console.log('to', to)
```
Comment (natoverse) [inline packages/react/src/compounds/FilterAggregateLookup/FilterAggregateLookup.tsx:72]: remove console.log

File: packages/react/src/controls/MergeInputs/MergeInputs.tsx [hunk 4]
```diff
@@ -24,6 +24,7 @@ export const MergeInputs: React.FC<StepComponentProps> = memo(
 			(event?: React.FormEvent<HTMLDivElement>, option?: IDropdownOption) => {
 				const { columns = [] } = internal.args
 				let update = [...columns]
+				console.log(event)
```
Comment (natoverse) [inline packages/react/src/controls/MergeInputs/MergeInputs.tsx:27]: remove console.log

**Review by natoverse — APPROVED** [review]

**Pull request merged**